lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,488 @@
+            - name: ansible_vmware_tools_executable
+"""
+
+EXAMPLES = r'''
+# example vars.yml
+---
+ansible_connection: vmware_tools
+
+ansible_vmware_host: vcenter.example.com
+ansible_vmware_user: administrator@vsphere.local
+ansible_vmware_password: Secr3tP4ssw0rd!12
+ansible_vmware_validate_certs: no  # default is yes
+ansible_vmware_silence_tls_warnings: yes # default is yes
+
+# vCenter Connection VM Path Example
+ansible_vmware_guest_path: DATACENTER/vm/FOLDER/{{ inventory_hostname }}
+# ESXi Connection VM Path Example
+ansible_vmware_guest_path: ha-datacenter/vm/{{ inventory_hostname }}
+
+ansible_vmware_tools_user: root
+ansible_vmware_tools_password: MyR00tPassw0rD
+
+# if the target VM guest is Windows set the 'ansible_shell_type' to 'powershell'
+ansible_shell_type: powershell
+
+
+# example playbook_linux.yml
+---
+- name: Test VMware Tools Connection Plugin for Linux
+  hosts: linux
+  tasks:
+    - command: whoami
+
+    - ping:
+
+    - copy:
+        src: foo
+        dest: /home/user/foo
+
+    - fetch:
+        src: /home/user/foo
+        dest: linux-foo
+        flat: yes
+
+    - file:
+        path: /home/user/foo
+        state: absent
+
+
+# example playbook_windows.yml
+---
+- name: Test VMware Tools Connection Plugin for Windows
+  hosts: windows
+  tasks:
+    - win_command: whoami
+
+    - win_ping:
+
+    - win_copy:
+        src: foo
+        dest: C:\Users\user\foo
+
+    - fetch:
+        src: C:\Users\user\foo
+        dest: windows-foo
+        flat: yes
+
+    - win_file:
+        path: C:\Users\user\foo
+        state: absent
+'''
+
+
+class Connection(ConnectionBase):
+    """VMware Tools Connection."""
+
+    transport = "vmware_tools"
+
+    @property
+    def vmware_host(self):
+        """Read-only property holding the connection address."""
+        return self.get_option("vmware_host")
+
+    @property
+    def validate_certs(self):
+        """Read-only property holding whether the connection should validate certs."""
+        return self.get_option("validate_certs")
+
+    @property
+    def authManager(self):
+        """Guest Authentication Manager."""
+        return self._si.content.guestOperationsManager.authManager
+
+    @property
+    def fileManager(self):
+        """Guest File Manager."""
+        return self._si.content.guestOperationsManager.fileManager
+
+    @property
+    def processManager(self):
+        """Guest Process Manager."""
+        return self._si.content.guestOperationsManager.processManager
+
+    @property
+    def windowsGuest(self):
+        """Return if VM guest family is windows."""
+        return self.vm.guest.guestFamily == "windowsGuest"
+
+    def __init__(self, *args, **kwargs):
+        """init."""
+        super(Connection, self).__init__(*args, **kwargs)
+        if hasattr(self, "_shell") and self._shell.SHELL_FAMILY == "powershell":
+            self.module_implementation_preferences = (".ps1", ".exe", "")
+            self.become_methods = ["runas"]
```

Comment:
these settings make this a 'windows only' connection , but examples above suggest otherwise, which is it?

Reply:
It supports both.  If SHELL_FAMILY=powershell, it does this special magic.  If not, it "just works" for Linux otherwise.